src/components/switch/switch.test.tsx

```diff
@@ -0,0 +1,357 @@
+
+  rerender(<Switch checked={false} onChange={onChangeMock} />);
+  expect(switchElement).not.toBeChecked();
+});
+
+test("has default a translation for on", () => {
+  render(<Switch checked onChange={jest.fn()} />);
+
+  const sliderPanelText = screen.getByText("ON");
+
+  expect(sliderPanelText).toBeVisible();
+});
+
+test("has default a translation for off", () => {
+  render(<Switch checked={false} onChange={jest.fn()} />);
+
+  const sliderPanelText = screen.getAllByText("OFF");
+
+  expect(sliderPanelText[1]).toBeVisible();
+});
+
+// required for styling coverage
+test("when `reverse` is false, the correct Label component styles are applied", () => {
+  render(<Switch reverse={false} label="label" />);
+
+  const labelContainer = screen.getByTestId("label-container");
+
+  expect(labelContainer).toHaveStyle("margin-bottom: 8px");
+});
+
+test("when `reverse` is true, the correct Label component styles are applied", () => {
+  render(<Switch reverse={false} label="label" />);
+
+  const labelContainer = screen.getByTestId("label-container");
+
+  expect(labelContainer).toHaveStyle("margin-top: 8px");
+});
+
+test("when `reverse` is false and `fieldHelpInline` is true, the correct FieldHelp component styles are applied", () => {
+  render(
+    <Switch
+      reverse={false}
+      fieldHelp="This text provides help"
+      fieldHelpInline
+      label="label"
+    />
+  );
+
+  const fieldHelp = screen.getByText("This text provides help");
+
+  expect(fieldHelp).toHaveStyle("margin-top: 8px");
+});
+
+test("when `labelInline` is true, fieldHelpInline is false and `reverse` is false, the correct FieldHelp component styles are applied", () => {
+  render(
+    <Switch
+      label="label"
+      fieldHelp="This text provides help"
+      fieldHelpInline={false}
+      labelInline
+      reverse={false}
+    />
+  );
+  const fieldHelp = screen.getByText("This text provides help");
+
+  expect(fieldHelp).toHaveStyle("margin-left: 60px");
+});
+
+test("when `fieldHelpInline` is true, the correct FieldHelp component styles are applied", () => {
+  render(
+    <Switch fieldHelpInline fieldHelp="This text provides help" label="label" />
+  );
+
+  const fieldHelp = screen.getByText("This text provides help");
+
+  expect(fieldHelp).toHaveStyle("margin: 0px");
+});
+
+test("when `labelInline` is true, the correct Label component styles are applied", () => {
+  render(<Switch label="label" labelInline />);
+
+  const labelContainer = screen.getByTestId("label-container");
+
+  expect(labelContainer).toHaveStyle("margin-bottom: 0px");
+});
+
+test("when `fieldHelpInline` is true and `labelInline` is true, the correct CheckableInput component styles are applied", () => {
+  render(
+    <Switch
+      label="label"
+      fieldHelp="This text provides help"
+      fieldHelpInline
+      labelInline
+    />
+  );
+
+  // eslint-disable-next-line testing-library/no-node-access
+  const checkableInput = screen.getByRole("switch").parentElement;
```

Comment:
I could not devise a better way of reaching this element. I am, however happy to take suggestions on this.

Reply:
Yeah. That seems far too obvious now you've suggested it 😶‍🌫️